File: src/main/java/com/mikea/bayes/belief/CliqueTreeQueryAlgorithm.java
package com.mikea.bayes.belief;

import com.mikea.bayes.BayesianNetwork;

/**
 * @author mike.aizatsky@gmail.com
 */
public class CliqueTreeQueryAlgorithm extends BeliefPropagationAlgorithm {

    public CliqueTreeQueryAlgorithm() {
        super(1);
    }

    @Override
    public BPResult run(BayesianNetwork network) {
        return new BPResult(network.getFactors(), CliqueTree.buildCliqueTree(network), iterations);
    }
}

File: src/main/java/com/mikea/bayes/belief/CliqueTreeQueryAlgorithm.scala
package com.mikea.bayes.belief

import com.mikea.bayes.BayesianNetwork
import com.mikea.bayes.belief.BeliefPropagationAlgorithm.BPResult

/**
 * @author mike.aizatsky@gmail.com
 */
class CliqueTreeQueryAlgorithm(iterations: Int) extends BeliefPropagationAlgorithm(iterations) {

  override def prepare(network: BayesianNetwork): BPResult = ???/*{
    new BPResult(network.getFactors, CliqueTree.buildCliqueTree(network), iterations)
  }*/
}
